dldashboard/web/design_doc.md

```diff
@@ -0,0 +1,41 @@
+A high level design diagram of the current architecture of the dashboard is :
+
+![dashboard architecture diagram](https://craftassist.s3-us-west-2.amazonaws.com/pubr/backend+architecture.png)
+
+In the diagram above:
+- frontend: The React based frontend that hosts other React subcomponents.
+- StateManager: This module manages messaging between frontend and backend. We set the state of the frontend components through the data received socket.io connections between stateManager and backend. 
+- backend : This is the agent process that sends and receives data from frontend. The agent process and its modules (perception, memory, mover, controller etc) can communicate using the the websocket transport.
+
+
+Here's a flowchart of how the current dashboard works:
+![dashboard flowchart](https://craftassist.s3-us-west-2.amazonaws.com/pubr/dashboard_flowchart.png)
```

Comment:
this is too dense to be useful imo, and it also won't date well. I'd consider just showing the lifecycle (in the form of a GIF) of the different kinds of components we have - ones that just read state (for eg Memory 2D, live image) and ones that write state (eg interact app).

Reply:
I think a GIF is an overkill for this. I can separate out reads from writes and make those two different diagrams.

Reply:
On a different note maybe we should put these in a table instead of a diagram.
Something like : 
```
Component name | Purpose | Socket Event | Read / Write operation from backend 
```

Reply:
yeah, let's have the components listed as a table and get rid of the diagram altogether.